pkg/backup/s3/s3.go

```diff
@@ -38,40 +38,36 @@ type S3 struct {
 // New returns a S3 translator from default shared config.
 // Please refer to http://docs.aws.amazon.com/sdk-for-go/api/aws/session/
 // for setting up credentials and configuration.
-func New(bucket, prefix string) (*S3, error) {
-	return NewFromSessionOpt(bucket, prefix, session.Options{
+func New(bucket, pathPrefix, prefix string) (*S3, error) {
+	return NewFromSessionOpt(bucket, pathPrefix, prefix, session.Options{
 		// Setting this is equal to the AWS_SDK_LOAD_CONFIG environment variable was set.
 		// We want to save the work to set AWS_SDK_LOAD_CONFIG=1 outside.
 		SharedConfigState: session.SharedConfigEnable,
 	})
 }
 
-func NewFromSessionOpt(bucket, prefix string, so session.Options) (*S3, error) {
+func NewFromSessionOpt(bucket, pathPrefix, prefix string, so session.Options) (*S3, error) {
 	sess, err := session.NewSessionWithOptions(so)
 	if err != nil {
 		return nil, fmt.Errorf("new AWS session failed: %v", err)
 	}
 	cli := s3.New(sess)
 
-	return &S3{
-		bucket: bucket,
-		prefix: prefix,
-		client: cli,
-	}, nil
+	return NewFromClient(bucket, pathPrefix, prefix, cli), nil
 }
 
-func NewFromClient(bucket, prefix string, cli *s3.S3) *S3 {
+func NewFromClient(bucket, pathPrefix, prefix string, cli *s3.S3) *S3 {
 	return &S3{
 		bucket: bucket,
-		prefix: prefix,
+		prefix: path.Join(pathPrefix, v1, prefix),
```

Comment:
It's confusing to have two "prefix" here.
It would be better to have a util Method:
```
func MakePrefix(s3path, clusterName, namespace string) string
```
And have external call it.

Inside the "pkg/backup/s3" layer it should be agnostic about any S3Path, cluster, namespace concept.